app/models/application_form.rb

```diff
@@ -356,6 +356,28 @@ def domicile
     end
   end
 
+  def mark_sections_incomplete_if_review_needed!
+    if becoming_a_teacher_reviewable?
+      update!(becoming_a_teacher_completed: false)
+    end
+  end
+
+  def becoming_a_teacher_rejection_reasons
+    CandidateInterface::RejectionReasonsHistory.all_previous_applications(self, :becoming_a_teacher)
+  end
+
+  def becoming_a_teacher_most_recent_rejection_reason
+    CandidateInterface::RejectionReasonsHistory.most_recent(self, :becoming_a_teacher)
```

Comment:
Just realised these method names are a bit misleading — it's the most recent application, not the most recent rejection reason (which may be from the application before last, for example).

Going to rename these to:
`becoming_a_teacher_previous_application_rejection_reason`
`RejectionResaonsHistory.previous_application`